File: challenges/ruby/HappyNumber/HappyNumber.rb
load "inc.rb"

$count = 0

def TesteNumberIsHappy(number)

    $count += 1

    des = LogicHappyNumber.new("Tentativa #{$count}, número #{number}\n\n")

    expected = true
    returned = des.HappyNumber(number)

    if expected != returned
        puts "Error , expected #{expected}, but returned #{returned} .\n\n"
    else
        puts "test is Ok !!! \n\n"
    end

end


TesteNumberIsHappy(100)
#TesteNumberIsHappy(7)
TesteNumberIsHappy(4)
#TesteNumberIsHappy(19)
#TesteNumberIsHappy(21)
puts "Insira um número para ser realizado o teste : "
number = gets()
TesteNumberIsHappy(number.to_i)

File: challenges/python/HappyNumber.py
def LogicHappyNumber (number, _list = []):

    if number == 1:
        return True
    if number in _list:
        return False
    else:
        acc = 0
        number = str(number)
        for i in number :
            acc += int(i) ** 2
            _list.append(int(i))
        if acc == 1:
            return True
        else:
            print(_list)
            return LogicHappyNumber(acc, _list)



def TesteNumberIsHappy(number):

    n_expected = True
    n_returned = bool(LogicHappyNumber(number))
    if n_expected != n_returned:
        print("Fail , expected {} but returned {} ".format(n_expected, n_returned))
    else:
        print("Test Ok, number is happy")




number = int(input("insira um número para ser realizado o teste : "))
TesteNumberIsHappy(number)
